ambassador/ambassador/ir/irauth.py

```diff
@@ -115,6 +115,13 @@ def _load_auth(self, module: Resource):
                     self[key] = value
 
             self.referenced_by(module)
+        
+        if module.get("add_linkerd_headers"):
+            self["add_linkerd_headers"] = module.get("add_linkerd_headers")
+        else:
+            add_linkerd_headers = module.get('add_linkerd_headers', None)
+            if add_linkerd_headers is None:
+                self["add_linkerd_headers"] = ir.ambassador_module.get('add_linkerd_headers', False)
```

Comment:
This logic, I think, is wrong if `add_linkerd_headers` is explicitly set to `False` in the `AuthService` when it's set to `True` in the `ambassador` `Module`. The logic below for `IRHTTPMapping` looks correct.